drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -78,8 +78,8 @@ static int _init(netdev2_t *netdev)
     gpio_set(dev->params.reset_pin);
     gpio_init_int(dev->params.int_pin, GPIO_IN, GPIO_RISING, _irq_handler, dev);
 
-    /* make sure device is not sleeping, so we can query part number */
-    at86rf2xx_assert_awake(dev);
+    /* reset device to default values and put it into RX state */
+    at86rf2xx_reset(dev);
```

Comment:
I forgot to mention this part. Previously, dev->state was set to SLEEP but after twiddling the gpio above it is actually in P_ON. If we go into assert_awake, that state mismatch means it waits for it to go into TRX_OFF but it won't because it is coming from P_ON not SLEEP. Assuming the chip state here was incorrect anyway, you should reset it before doing any communication.